lib/cronofy/client.rb

```diff
@@ -1492,6 +1492,69 @@ def delete_availability_rule(availability_rule_id)
       nil
     end
 
+    # Public: Creates or updates an AvailablePeriod.
+    #
+    # available_period_id - A String uniquely identifying the available period
+    #                       for the authenticated user in your application
+    #                       (note: this is NOT an ID generated by Cronofy).
+    # body                - A Hash describing the available period with
+    #                       symbolized keys:
+    #                       :start - A String (ISO-8601 date/time)
+    #                       :end   - A String (ISO-8601 date/time)
+    #
+    # See https://docs.cronofy.com/developers/api/scheduling/available-periods/upsert/
+    # for reference.
+    #
+    # Returns nothing.
+    #
+    # Raises Cronofy::CredentialsMissingError if no credentials available.
+    # Raises Cronofy::InvalidRequestError if the request contains invalid
+    # parameters.
+    # Raises Cronofy::TooManyRequestsError if the request exceeds the rate
+    # limits for the application.
+    def upsert_available_period(available_period_id, body)
+      payload = body.merge(available_period_id: available_period_id)
+      wrapped_request { post("/v1/available_periods", payload) }
+      nil
+    end
+
+    # Public: Gets all AvailablePeriods for an account.
+    #
+    # Returns an array of AvailablePeriods.
+    #
+    # Raises Cronofy::CredentialsMissingError if no credentials available.
+    # Raises Cronofy::InvalidRequestError if the request contains invalid
+    # parameters.
+    # Raises Cronofy::TooManyRequestsError if the request exceeds the rate
+    # limits for the application.
+    def get_available_periods(options={})
+      query = [
+        "from=#{to_iso8601(options[:from])}",
+        "to=#{to_iso8601(options[:to])}",
+        "tzid=#{options[:tzid]}"
```

Comment:
Looks like there's also a "localized_times" option? https://docs.cronofy.com/developers/api/scheduling/available-periods/read/#param-localized_times

Reply:
We aren't using that parameter. Support could be added in a separate PR.